### PR TITLE
[aspire]add known param - container env name

### PR DESCRIPTION
**Pull request by vhvb1989**
- Adding new known parameter for Aspire bicep modules injection: ContainerEnvName
- snaps for functional test

fix: https://github.com/Azure/azure-dev/issues/3931

### CODE REVIEW

File: cli/azd/test/functional/aspire_test.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func restoreDotnetWorkload(t *testing.T) {
 
 // Test_CLI_Aspire_DetectGen tests the detection and generation of an Aspire project.
 func Test_CLI_Aspire_DetectGen(t *testing.T) {
-	restoreDotnetWorkload(t)
+	//restoreDotnetWorkload(t)
```
Comment (davidfowl) [inline cli/azd/test/functional/aspire_test.go:71]: ?

File: cli/azd/pkg/apphost/generate.go [hunk 2]
```diff
@@ -778,6 +780,9 @@ func injectValueForBicepParameter(resourceName, p string, parameter any) (string
 	if p == knownParameterLogAnalytics {
 		return knownInjectedValueLogAnalytics, true, nil
 	}
+	if p == knownParameterContainerEnvName {
```
Comment (davidfowl) [inline cli/azd/pkg/apphost/generate.go:783]: Lets add the containerAppEnvironmentId as well

File: cli/azd/test/functional/aspire_test.go (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/test/functional/aspire_test.go]: unwanted... I usually need to do this to re-create test snapshots...

File: cli/azd/pkg/apphost/generate.go (hunk 2 above)
Comment (vhvb1989) [reply cli/azd/pkg/apphost/generate.go:785]: added

**Comment by davidfowl** [comment]
Works like a charm! https://github.com/davidfowl/EventGridDemo/blob/f02811a004d41e71219f854afef47bc2b566c8db/EventGridDemo.AppHost/EventGridWebHookResource.cs#L28

File: cli/azd/test/functional/aspire_test.go (hunk 1 above)
Comment (ellismg) [reply cli/azd/test/functional/aspire_test.go]: You should get on my sudo plan, it's actually really nice!  I just need to set the `AZD_TEST_DOTNET_WORKLOAD_USE_SUDO` and it is nice.

Comment (vhvb1989) [reply cli/azd/test/functional/aspire_test.go]: I did, but it didn't help me for using the UI `run test` button.  Did you figure a way to make that one to work?

image


The `USE_SUDO` hack worked for invoking the test from the console :)...

**Comment by davidfowl** [comment]
> This is a nice primitive to have to be able to unblock some of the EventGrid stuff, but I'm excited also to figure out a more structured way to solve this problem long term that feels a bit more declarative vs the need to deploy a dummy app during provision.

Agreed. This is a mega hack 😄, but at least it's not a hack in azd. Just my repo. What I would have preferred to do was to have a dependency between the container app and the event grid resource so that we could make this work, but that is a longer discussion entangled with compute customization.

**Pull request merged**